test/core/metrics/metrics_test.cpp

```diff
@@ -0,0 +1,321 @@
+#include "metrics/metrics.hpp"
+#include <gtest/gtest.h>
+#include <cmath>
+#include <thread>
+#include "metrics/impl/prometheus/registry_impl.hpp"
+
+template <typename T>
+prometheus::ClientMetric getMetric(T *metric) {
+  return kagome::metrics::PrometheusRegistry::internalMetric(metric)->Collect();
+}
+
+class CounterTest : public ::testing::Test {
+  kagome::metrics::RegistryPtr registry_;
+
+ public:
+  void SetUp() override {
+    registry_ = kagome::metrics::createRegistry();
+  }
+
+  kagome::metrics::Counter *createCounter(const std::string &name) {
+    registry_->registerCounterFamily(name);
+    return registry_->registerCounterMetric(name);
+  }
+};
+
+// all tests have same static prometheus::Registry
+
+TEST_F(CounterTest, initialize_with_zero) {
```

Comment:
Please write given-when-then explanation for tests when possible. Example: https://github.com/soramitsu/kagome/blob/a407ae07bb6d8cd01c89c2a2e063caf317bc8bf3/test/core/application/app_config_test.cpp#L299

Reply:
+1

Reply:
Not an issue. We write test case names using CamelCase too, as for fixture names.